tests/integrationtest/l4_service_test.go

```diff
@@ -423,22 +423,30 @@ func TestCreateServiceLBWithFaultCacheSync(t *testing.T) {
 		var finalResponse []byte
 		url := r.URL.EscapedPath()
 
-		if strings.Contains(url, "macro") && r.Method == "POST" {
+		rModelName := ""
+		if r.Method == "POST" && !strings.Contains(url, "login") {
 			data, _ := ioutil.ReadAll(r.Body)
 			json.Unmarshal(data, &resp)
-			rData, rModelName := resp["data"].(map[string]interface{}), strings.ToLower(resp["model_name"].(string))
-			if rModelName == "virtualservice" && injectFault {
+			fmt.Printf("xxx: resp: %v", resp)
+			if strings.Contains(url, "virtualservice") && injectFault {
 				injectFault = false
 				w.WriteHeader(http.StatusBadRequest)
 				fmt.Fprintln(w, `{"error": "bad request"}`)
 			} else {
-				rName := rData["name"].(string)
+				if strings.Contains(url, "virtualservice") {
+					rModelName = "virtualservice"
+				} else if strings.Contains(url, "vsvip") {
+					rModelName = "vsvip"
+				} else if strings.Contains(url, "l4policy") {
+					rModelName = "l4policy"
```

Comment:
Can we call it the full name viz. l4policyset?

Reply:
Done